soc/nxp/common/Kconfig.flexspi_xip

```diff
@@ -0,0 +1,41 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+DT_CHOSEN_Z_FLASH := zephyr,flash
+DT_COMPAT_FLEXSPI := nxp,imx-flexspi
+
+DT_CHOSEN_FLASH_NODE := $(dt_chosen_path,$(DT_CHOSEN_Z_FLASH))
+DT_CHOSEN_FLASH_PARENT := $(dt_node_parent,$(DT_CHOSEN_FLASH_NODE))
+
+DT_FLASH_PARENT_IS_FLEXSPI := $(dt_node_has_compat,$(DT_CHOSEN_FLASH_PARENT),$(DT_COMPAT_FLEXSPI))
+DT_FLASH_HAS_SIZE_PROP := $(dt_node_has_prop,$(DT_CHOSEN_FLASH_NODE),size)
+
+config FLASH_BASE_ADDRESS
+	default $(dt_node_reg_addr_hex,$(DT_CHOSEN_FLASH_PARENT),1) \
+		if $(DT_FLASH_PARENT_IS_FLEXSPI)
+	default $(dt_chosen_reg_addr_hex,$(DT_CHOSEN_Z_FLASH))
+
+config FLASH_SIZE
+	default $(dt_node_int_prop_int,$(DT_CHOSEN_FLASH_NODE),size,Kb) \
+		if $(DT_FLASH_HAS_SIZE_PROP)
+	default $(dt_chosen_reg_size_int,$(DT_CHOSEN_Z_FLASH),0,K)
+
+config FLASH_MCUX_FLEXSPI_XIP
+	bool
+	default $(DT_FLASH_PARENT_IS_FLEXSPI)
+	select XIP
+	help
+	  Allows REfor the soc to safely initialize the clocks for the
+	  FlexSpi when planning to execute code in FlexSpi Memory.
+
+config CODE_DATA_RELOCATION_SRAM
+	default y if FLASH_MCUX_FLEXSPI_XIP && MEMC_MCUX_FLEXSPI
```

Comment:
I *think* the dependency here is on the flash driver, not on the MEMC driver. For example on the RT595 we can use the memc driver while running in XIP mode to interface with HyperRAM on the second FlexSPI device, and we don't need to relocate code to SRAM

Reply:
this is what the dependency was before this PR, I just moved it to a different file, while you might be right, I don't want to attempt this change in this PR